hypervisor/arch/x86/guest/vioapic.c

```diff
@@ -393,14 +393,10 @@ vioapic_write(struct vioapic *vioapic, uint32_t addr, uint32_t data)
 			vioapic_send_intr(vioapic, pin);
 		}
 
-		/* remap for active: interrupt mask -> unmask
-		 * remap for deactive: interrupt mask & vector set to 0
+		/*
+		 * remap for any update to RTE
 		 */
-		data64 = vioapic->rtbl[pin].reg;
-		if ((((data64 & IOAPIC_RTE_INTMASK) == IOAPIC_RTE_INTMCLR)
-		  && ((last & IOAPIC_RTE_INTMASK) == IOAPIC_RTE_INTMSET))
-		  || (((data64 & IOAPIC_RTE_INTMASK) == IOAPIC_RTE_INTMSET)
-		  && ((vioapic->rtbl[pin].reg & IOAPIC_RTE_INTVEC) == 0))) {
```

Comment:
we want to keep the filter,  and would not like to trigger ptdev remapping for every change. Could we think about what field need be reflect to native, and just add them into check filter?
Like here we can add  IOAPIC_RTE_TRGRMOD, IOAPIC_RTE_INTPOL, IOAPIC_RTE_DESTMOD, IOAPIC_RTE_DELMOD and IOAPIC_RTE_DEST.

Reply:
Yes we can add the fields. For this specific case being discussed, only IOAPIC_RTE_TRGRMOD and IOAPIC_RTE_INTPOL would suffice. But my original question remains what is the reason behind original filter of only mask change?
Related mailing list: https://lists.projectacrn.org/g/acrn-users/topic/question_regarding/22866607?p=,,,20,0,0,0::recentpostdate%2Fsticky,,,20,2,0,22866607

Reply:
@abdulrehman-MentG, usually, we should not change RTE setting during unmask, and for performance consideration, we don't want ptdev remapping run every time vioapic RTE got update. That's why we only take care mask change.